src/compressUtils.js

```diff
@@ -53,8 +52,24 @@ const uglifyOpts = {
 };
 
 function buildConnection(dir = cwd) {
+  const name = path.basename(dir);
   const connectionPath = path.resolve(dir, 'connection', 'connection.js');
   const connectionExports = require(connectionPath);
+  const ctxDef = connectionExports.contextDefinition;
+
+  const f = path.resolve(dir, `${name}.json`);
+  return fs.readJson(f)
```

Comment:
Any reason behind using the async version of (read/write)Json, unlike the sync ones used in the tests? I think, if we use the sync version, we can avoid promise chaining in ` bin/build*` files. We also will not have to wrap `checkConflictingCtxDefs` with a promise.

Reply:
Good point. I changed it to use sync methods for fs access. It's definitely cleaner this way.

@iamigo

Reply:
@jgraff2  Sorry. I did not realize that my comment would cause so many other changes. We could have done it as a part of another story too.